handler/src/test/java/io/netty/handler/ssl/SslHandlerTest.java

```diff
@@ -63,4 +67,52 @@ public void testNonByteBufNotPassThrough() throws Exception {
 
         ch.writeOutbound(new Object());
     }
+
+    private static class TlsReadTest extends ChannelOutboundHandlerAdapter {
+        public volatile boolean readIssued;
+
+        @Override
+        public void read(ChannelHandlerContext ctx) throws Exception {
+            readIssued = true;
+            super.read(ctx);
+        }
+
+        public void test(final boolean dropChannelActive) throws Exception {
+          SSLEngine engine = SSLContext.getDefault().createSSLEngine();
+          engine.setUseClientMode(true);
+
+          ChannelHandler[] handlers = {
+              this,
+              new SslHandler(engine),
+              new ChannelInboundHandlerAdapter() {
+                @Override
+                public void channelActive(ChannelHandlerContext ctx) throws Exception {
+                  if (!dropChannelActive) {
+                    ctx.fireChannelActive();
+                  }
+                }
+              }
+          };
```

Comment:
Why not just do this directly in the constructor of `EmbeddedChannel` and so not create the array ?